contracts/NFTVesting/ConvexVestingNFTTimeLock.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v4.5.0) (token/ERC20/utils/TokenTimelock.sol)
+// SpartanLabs Contracts (NFTVesting)
+
+pragma solidity ^0.8.0;
+import "./IERC721.sol";
+
+
+/**
+ * @dev A single NFT holder contract that will allow a beneficiary to extract the
+ * NFT after a given release time.
+ *
+ * Useful for simple vesting schedules like "whitelisted addresses get their NFT
+ * after 1 year".
+ */
+contract ConvexVestingNFTTimeLock {
+
+    // ERC721 basic token smart contract
+    IERC721 private immutable _nft;
+
+    // ERC721 basic token ID of contract being held
+    uint256 private immutable _tokenId;
+
+    // beneficiary of token after they are released
+    address private immutable _beneficiary;
+
+    // timestamp when token release is enabled and when discount starts to vest
+    uint256 private immutable _releaseTime;
+
+    // Max discount allowed for a token in percentage
+    uint256 private immutable _maxDiscount;
+
+    // Duration that token will vest
+    uint256 private _maxDuration;
+
+    // Growth rate for vesting. M in MX^n
+    uint256 private _growthRate;
+
+    // exponent for vesting
+    uint8 private _exponent;
+
+
+    /**
+     * @dev Deploys a timelock instance that is able to hold the token specified, and will only release it to
+     * `beneficiary_` when {release} is invoked after `releaseTime_`. The release time is specified as a Unix timestamp
+     * (in seconds).
+     */
+    constructor(
+        IERC721 nft_,
+        uint256 tokenId_,
+        address beneficiary_,
+        uint256 releaseTime_,
+        uint256 maxDiscount_,
+        uint256 maxDuration_,
+        uint256 growthRate_,
+        uint8 exponent_
+    ) {
```

Comment:
added reqirement > 0

Reply:
Does it also mean that we should have requirements for the NFT being owned by this address by this point for all contracts?

Reply:
I was initially thinking that in the front end, the user would have done the checks to make sure that NFT is being owned by this address, so that it allows for more flexibility.  Should we have it in our contract?